### DISCUSSION TITLE
Steps to Build / Compile plugin from Source

### DISCUSSION THREAD

**Discussion by Krizen2000** [brianpetro/obsidian-smart-connections#898] [General]
Hi,

I have been trying to build the plugin to debug issues using the Llama.cpp engine.

I cloned the repository and also cloned the jsbrains repository. I installed the Node.js modules using Node.js version 20.

However, the problem occurs when I try to run build, and this error pops up.

![image](https://github.com/user-attachments/assets/c2a3897c-258f-42db-baf5-3338cad48b42)

I have already installed the packages in the jsbrains repository, but I’m still encountering this issue.

**Comment by brianpetro** [comment]
Hi @Krizen2000 

Make sure you have the [JS Brains repo](https://github.com/brianpetro/jsbrains) in a directory adjacent to the obsidian-smart-connections folder.

You'll also need to make sure that `npm install` is run in each of the sub-folders in the `jsbrains/` directory.

🌴

**Reply by Krizen2000** [reply]
Hi @brianpetro 

I ran `npm install` in the sub-folders but am still getting the same error. Yes I have kept the **jsbrains** directory adjacent to the **obsidian-smart-connections** folder.

**Reply by Krizen2000** [reply]
It seems doing `npm install --include=dev` in all sub-folder saved the day. Thanks!

**Reply by brianpetro** [reply]
@Krizen2000 great, thanks for following up with the solution 😊🌴
